hubblestack/extmods/modules/nova.py

```diff
@@ -162,20 +181,23 @@ def __execute_module(audit_id, audit_impl, audit_data, verbose):
     for audit_check in audit_impl['checks']:
         __nova__[validate_param_method](audit_id, audit_check)
 
+    #check for type in check implementation. If not present default is 'and'
+    type = audit_impl.get('type', 'and').lower().strip()
+    audit_result['type']=type
     # validate succeded, lets execute it and prepare result dictionary
     audit_result['run_config']['checks'] = []
     execute_method = audit_impl['module'] + '.execute'
     filtered_log_method = audit_impl['module'] + '.get_filtered_params_to_log'
+    overall_result = type=='and'
+
     for audit_check in audit_impl['checks']:
         module_result_local = __nova__[execute_method](audit_id, audit_check)
         audit_result_local = {}
         if module_result_local['result']:
             audit_result_local['check_result'] = CHECK_STATUS['Success']
         else:
-            # if 'failure_reason' in audit_data:
-            #     audit_result_local
             audit_result_local['check_result'] = CHECK_STATUS['Failure']
-        
+            audit_result_local['failure_reason'] = module_result_local['failure_reason']
```

Comment:
In profile schema, we have introduced this attribute: "failure_schema"
If it is present, that should override any failure reason value from modue.

Reply:
Done